scripts/validate_rst_title_capitalization.py

```diff
@@ -255,10 +268,12 @@ def main(source_paths: list[str]) -> int:
 
     for filename in source_paths:
         for title, line_number in find_titles(filename):
-            if title != correct_title_capitalization(title):
+            removed_https_title = re.sub(r"<https?:\/\/.*[\r\n]*>", "", title)
+            if removed_https_title != correct_title_capitalization(removed_https_title):
                 print(
-                    f"""{filename}:{line_number}:{err_msg} "{title}" to "{
-                    correct_title_capitalization(title)}" """
+                    f"""{filename}:{line_number}:{err_msg} "{
+                    removed_https_title.strip()}" to "{
+                    correct_title_capitalization(removed_https_title).strip()}" """
```

Comment:
In a second thought, I'd probably just force all urls to be lowercase. The code will be simpler, and it's probably a good idea anyway, even if some people make them follow their branding, as urls are case insensitive.